soc/arm/microchip_mec/common/spigen/mec_spi_gen.py

```diff
@@ -0,0 +1,498 @@
+        hdr[5] |= HDR_SPI_CPOL_HI
+    if spi['spi_mode'] & 0x02:
+        hdr[5] |= HDR_SPI_CHPHA_MOSI_EDGE_1
+    if spi['spi_mode'] & 0x04:
+        hdr[5] |= HDR_SPI_CHPHA_MISO_EDGE_2
+
+    # translate 1x, 2x, 4x, 6x to 0, 1, 2, 3
+    if spi['spi_drive_str'] == "6x":
+        hdr[5] |= HDR_SPI_DRV_STR_6X
+    elif spi['spi_drive_str'] == "4x":
+        hdr[5] |= HDR_SPI_DRV_STR_4X
+    elif spi['spi_drive_str'] == "2x":
+        hdr[5] |= HDR_SPI_DRV_STR_2X
+    else:
+        hdr[5] |= HDR_SPI_DRV_STR_1X
+
+    # translate "slow", "fast" to 0, 1
+    if spi['spi_slew_rate'] == "fast":
+        hdr[5] |= HDR_SPI_SLEW_FAST
+
+    # MEC172x b[0]=0 do not allow 96MHz SPI clock
+    hdr[6] = 0 # not using authentication or encryption
+
+    if spi['spi_read_mode'] == 'quad':
+        hdr[7] = HDR_SPI_RD_1148
+    elif spi['spi_read_mode'] == 'dual':
+        hdr[7] = HDR_SPI_RD_1128
+    elif spi['spi_read_mode'] == 'normal':
+        hdr[7] = HDR_SPI_RD_111
+    else:
+        hdr[7] = HDR_SPI_RD_1118
+
+    # payload load address in SRAM
+    pld_load_addr = CHIP_DICT[chip]['sram_base']
+    hdr[8] = pld_load_addr & 0xff
+    hdr[9] = (pld_load_addr >> 8) & 0xff
+    hdr[0xA] = (pld_load_addr >> 16) & 0xff
+    hdr[0xB] = (pld_load_addr >> 24) & 0xff
+
+    # payload entry point address in SRAM
+    hdr[0xC] = pld_entry_addr & 0xff
+    hdr[0xD] = (pld_entry_addr >> 8) & 0xff
+    hdr[0xE] = (pld_entry_addr >> 16) & 0xff
+    hdr[0xF] = (pld_entry_addr >> 24) & 0xff
+
+    # payload size (16-bit) in granularity units
+    pld_units = pld_len // PLD_GRANULARITY
+    hdr[0x10] = pld_units & 0xff
+    hdr[0x11] = (pld_units >> 8) & 0xff
+    # hdr[0x12:0x13] = 0 reserved
+
+    # Unsigned offset from start of Header to start of FW Binary
+    # FW binary(payload) must always be located after header
+    pld_offset = pld_spi_loc - hdr_spi_loc
+    hdr[0x14] = pld_offset & 0xff
+    hdr[0x15] = (pld_offset >> 8) & 0xff
+    hdr[0x16] = (pld_offset >> 16) & 0xff
+    hdr[0x17] = (pld_offset >> 24) & 0xff
+
+    # hdr[0x18] = 0 not using authentication
+    # hdr[0x19] = 0 not adjusting SPI flash device drive strength
+    # hdr[0x1A through 0x1F] = 0 reserved
+    # hdr[0x20 through 0x27] = 0 not adjust SPI flash device drive strength
+    # hdr[0x28 through 0x47] = 0 reserved
+    # hdr[0x48 through 0x4F] = 0 reserved
+    # hdr[0x50 through 0x7F] = ECDSA P-384 Public key x-component
+    # hdr[0x80 through 0xAF] = ECDSA P-384 Public key y-component
+    # hdr[0xB0 through 0xDF] = SHA-384 digest of hdr[0 through 0xAF] Always required
+    # hdr[0xE0 through 0x10F] = ECDSA signature R-component of hdr[0 through 0xDF]
+    # hdr[0x110 through 0x13F] = ECDSA signature S-component of hdr[0 through 0xDF]
+
+    h = hashlib.sha384()
+    h.update(hdr[0:0xB0])
+    hdr_digest = h.digest()
+
+    if VERBOSE_MODE:
+        print_bytes("hdr_sha384_digest", hdr_digest)
+
+    hdr[0xB0:0xE0] = hdr_digest
+
+    return bytes(hdr)
+# end build_header
```

Comment:
Drop these? Don't think any other Python script in the codebase uses these kind of notations.

Reply:
can you clarify? is the list slice an issue or the return statement?

Reply:
removed function closing comments